src/Symfony/Component/DependencyInjection/Dumper/PhpDumper.php

```diff
@@ -555,8 +556,12 @@ private function generateProxyClasses(): array
                 continue;
             }
             $alreadyGenerated[$asGhostObject][$class] = true;
-            // register class' reflector for resource tracking
-            $this->container->getReflectionClass($class);
+
+            $r = $this->container->getReflectionClass($class);
+            do {
+                $this->container->addResource(new FileResource($r->getFileName()));
```

Comment:
this should account for cases where `$r->getFilename()` does not return a file path (when it returns `false` for classes defined in C or when it returns a weird string for classes defined through `eval`)